### ISSUE TITLE
Could not install from update site using JDK 11

### ISSUE THREAD

**Issue by JochenHiller** [groovy/groovy-eclipse#823]
# How to reproduce

* download Eclipse JDT 2018-12
* start Eclipse using JDK 11
* add update site to https://dist.springsource.org/release/GRECLIPSE/e4.10/
* try to install from there, does not work with JDK 11
* do the same with JDK 8, you can install successfully

# More details

I tried to install GRECLIPSE e4.10 into a fresh Eclipse JDT 2018-12: it does not work with JDK 11.

It will report certificate exception below. 
Note: other p2 repos using https are working well

```
Unable to read repository at https://dist.springsource.org/release/GRECLIPSE/e4.10/compositeContent.xml.
Unable to read repository at https://dist.springsource.org/release/GRECLIPSE/e4.10/compositeContent.xml.
peer not authenticated

javax.net.ssl.SSLPeerUnverifiedException: peer not authenticated
	at java.base/sun.security.ssl.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:526)
	at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:113)
	at org.apache.http.conn.ssl.SSLSocketFactory.verifyHostname(SSLSocketFactory.java:580)
	at org.apache.http.conn.ssl.SSLSocketFactory.createLayeredSocket(SSLSocketFactory.java:574)
	at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:557)
	at org.eclipse.ecf.provider.filetransfer.httpclient4.SNIAwareHttpClient$1.connectSocket(SNIAwareHttpClient.java:64)
	at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:414)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.eclipse.ecf.provider.filetransfer.httpclient4.HttpClientFileSystemBrowser.runRequest(HttpClientFileSystemBrowser.java:263)
	at org.eclipse.ecf.provider.filetransfer.browse.AbstractFileSystemBrowser$DirectoryJob.run(AbstractFileSystemBrowser.java:69)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

I am using:

```
eclipse.buildId=4.10.0.I20181206-0815
java.version=11.0.2
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_GB
Framework arguments:  -product org.eclipse.epp.package.jee.product -keyring /Users/XXXXXX/.eclipse_keyring
Command-line arguments:  -os macosx -ws cocoa -arch x86_64 -product org.eclipse.epp.package.jee.product -keyring /Users/XXXXXX/.eclipse_keyring

Update site: https://dist.springsource.org/release/GRECLIPSE/e4.10
```

I tested the same with running Eclipse with JavaSE 8u201 and I can install it successfully. So I conclude it fails due to JDK 11.
On commandline I could download with a Java based curl from  https://dist.springsource.org/release/GRECLIPSE/e4.10/compositeContent.xml successfully with JDK 8 and JDK 11 successfully. So it seems to be somehow Eclipse and JDK 11 related?

**Comment by eric-milles** [comment]
Can you use http instead of https?

**Comment by JochenHiller** [comment]
No, does not work. If you do a 

```
curl --http1.1 -v -o someFile http://dist.springsource.org/release/GRECLIPSE/e4.10/compositeContent.xml
```

it will redirect you to `https://dist.springsource.org/release/GRECLIPSE/e4.10/compositeContent.xml`

In Eclipse I see exactly same error.

**Comment by eric-milles** [comment]
Pivotal is investigating.  Thanks for raising this issue.

**Comment by eric-milles** [comment]
Could you try again?  I'm told the certificates have been adjusted.

**Comment by JochenHiller** [comment]
No, does not work. I see same issue:

```
javax.net.ssl.SSLPeerUnverifiedException: peer not authenticated
	at java.base/sun.security.ssl.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:526)
	at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:113)
	at org.apache.http.conn.ssl.SSLSocketFactory.verifyHostname(SSLSocketFactory.java:580)
	at org.apache.http.conn.ssl.SSLSocketFactory.createLayeredSocket(SSLSocketFactory.java:574)
	at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:557)
	at org.eclipse.ecf.provider.filetransfer.httpclient4.SNIAwareHttpClient$1.connectSocket(SNIAwareHttpClient.java:64)
	at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:414)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.eclipse.ecf.provider.filetransfer.httpclient4.HttpClientFileSystemBrowser.runRequest(HttpClientFileSystemBrowser.java:263)
	at org.eclipse.ecf.provider.filetransfer.browse.AbstractFileSystemBrowser$DirectoryJob.run(AbstractFileSystemBrowser.java:69)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

eclipse.buildId=4.10.0.I20181206-0815
java.version=11.0.2
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_GB
Framework arguments:  -product org.eclipse.epp.package.jee.product -keyring /Users/xxx/.eclipse_keyring
Command-line arguments:  -os macosx -ws cocoa -arch x86_64 -product org.eclipse.epp.package.jee.product -keyring /Users/xxx/.eclipse_keyring
```

I am also NOT able to reproduce that via any commandline tests using  different tools, neither native curl or any Java based curl using JDK 11.
Please note that given to stacktrace Eclipse/ECF is internally using Apache HttpClient 4 using a SNIAware HttpClient. Maybe this is a root cause?

**Comment by aclement** [comment]
Can you install OK from the https link to the direct release referenced from that composite site, which is: `https://dist.springsource.org/release/GRECLIPSE/3.2.0/e4.10/`

**Comment by JochenHiller** [comment]
No does also not work. Same exceptions:

```
Unable to read repository at https://dist.springsource.org/release/GRECLIPSE/3.2.0/e4.10/compositeContent.xml.
Unable to read repository at https://dist.springsource.org/release/GRECLIPSE/3.2.0/e4.10/compositeContent.xml.
peer not authenticated

javax.net.ssl.SSLPeerUnverifiedException: peer not authenticated
	at java.base/sun.security.ssl.SSLSessionImpl.getPeerCertificates(SSLSessionImpl.java:526)
	at org.apache.http.conn.ssl.AbstractVerifier.verify(AbstractVerifier.java:113)
	at org.apache.http.conn.ssl.SSLSocketFactory.verifyHostname(SSLSocketFactory.java:580)
	at org.apache.http.conn.ssl.SSLSocketFactory.createLayeredSocket(SSLSocketFactory.java:574)
	at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:557)
	at org.eclipse.ecf.provider.filetransfer.httpclient4.SNIAwareHttpClient$1.connectSocket(SNIAwareHttpClient.java:64)
	at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:414)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:326)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:610)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:445)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.eclipse.ecf.provider.filetransfer.httpclient4.HttpClientFileSystemBrowser.runRequest(HttpClientFileSystemBrowser.java:263)
	at org.eclipse.ecf.provider.filetransfer.browse.AbstractFileSystemBrowser$DirectoryJob.run(AbstractFileSystemBrowser.java:69)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

eclipse.buildId=4.10.0.I20181206-0815
java.version=11.0.2
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_GB
Framework arguments:  -product org.eclipse.epp.package.jee.product -keyring /Users/jhiller/.eclipse_keyring
Command-line arguments:  -os macosx -ws cocoa -arch x86_64 -product org.eclipse.epp.package.jee.product -keyring /Users/jhiller/.eclipse_keyring
```

**Comment by aclement** [comment]
Some settings on the servers have been tweaked. This is working for me now (using JDK11, installing from https://dist.springsource.org/release/GRECLIPSE/e4.10/ )

**Comment by JochenHiller** [comment]
Hi Andy,
great: it does work for me as well. Thanks for your help.

Can you give more details what you "tweaked" at Pivotal for the cloud servers? It might be of interest if other people seeing similar issues running JDK 11 and Apache HttpClient/Eclipse ECF.

**Comment by aclement** [comment]
As I understand it, turned off tls1.3: https://bugs.openjdk.java.net/browse/JDK-8211806

**Comment by JochenHiller** [comment]
Thanks Andy.
Interesting link. But they say that it has been fixed in 11.0.2, which I used for testing. So there must be another reason, maybe related?

Nevertheless thanks for you help.
